src/backend/sev/snp/launch/mod.rs

```diff
@@ -90,7 +95,15 @@ impl<U: AsRawFd, V: AsRawFd> Launcher<Started, U, V> {
         let launch_update_data = LaunchUpdate::from(update);
         let mut cmd = Command::from(&mut self.sev, &launch_update_data);
 
-        KvmEncRegion::new(update.uaddr).register(&mut self.vm_fd)?;
+        let kvm = Kvm::new()?;
+        let vm_fd = unsafe { libc::dup(self.vm_fd.as_raw_fd()) };
+        let vm = unsafe { kvm.create_vmfd_from_rawfd(vm_fd)? };
+
+        let memory_region = kvm_enc_region {
+            addr: update.uaddr.as_ptr() as _,
+            size: update.uaddr.len() as _,
+        };
+        vm_fd.register_enc_memory_region(&memory_region)?;
```

Comment:
```suggestion
        let memory_region = kvm_enc_region {
            addr: update.uaddr.as_ptr() as _,
            size: update.uaddr.len() as _,
        };
        self.vm_fd.register_enc_memory_region(&memory_region)?;
```

Reply:
I did it how I did it because doing what you suggested resulted:
```
error[E0599]: no method named `register_enc_memory_region` found for type parameter `U` in the current scope
   --> src/backend/sev/snp/launch/mod.rs:102:20
    |
102 |         self.vm_fd.register_enc_memory_region(&memory_region)?;
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `U`
```

Reply:
This has been fixed in #1250 by refactoring struct Launcher first.